plugins/database/quota/src/org/apache/cloudstack/api/command/QuotaCreditsCmd.java

```
@@ -0,0 +1,147 @@
+    private Boolean quotaEnforce;
+
+    public Double getMinBalance() {
+        return minBalance;
+    }
+
+    public void setMinBalance(Double minBalance) {
+        this.minBalance = minBalance;
+    }
+
+    public Boolean getQuotaEnforce() {
+        return quotaEnforce;
+    }
+
+    public void setQuotaEnforce(Boolean quotaEnforce) {
+        this.quotaEnforce = quotaEnforce;
+    }
+
+    public String getAccountName() {
+        return accountName;
+    }
+
+    public void setAccountName(String accountName) {
+        this.accountName = accountName;
+    }
+
+    public Long getDomainId() {
+        return domainId;
+    }
+
+    public void setDomainId(Long domainId) {
+        this.domainId = domainId;
+    }
+
+    public Double getValue() {
+        return value;
+    }
+
+    public void setValue(Double value) {
+        this.value = value;
+    }
+
+    public QuotaCreditsCmd() {
+        super();
+    }
+
+    @Override
+    public String getCommandName() {
+        return s_name;
+    }
+
+    @Override
+    public void execute() {
+        Long accountId = null;
+        Account account = _accountService.getActiveAccountByName(accountName, domainId);
+        if (account != null) {
+            accountId = account.getAccountId();
+        }
+        if (accountId == null) {
+            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "The account does not exists or has been removed/disabled");
+        }
+        if (getValue() == null) {
+            throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Please send a valid non-empty quota value");
+        }
+        if (getQuotaEnforce() != null) {
+            _quotaService.setLockAccount(accountId, getQuotaEnforce());
+        }
+        if (getMinBalance() != null) {
+            _quotaService.setMinBalance(accountId, getMinBalance());
+        }
+        else {
+            _quotaService.setMinBalance(accountId, 0.2 * getValue());
```

We should throw an exception if misbalance is not set, instead of using an arbitrary default value.